### DISCUSSION TITLE
QUESTION Custom CSS to Twitch Chat on OBS

### DISCUSSION THREAD

**Discussion by KATalyzt75** [obsproject/obs-studio#5332] [Development Questions] [answered]
Is possible add my own CSS file to the chat? (Like BTTV do with him)
This file only try to change color to /me messages for my bot

**Comment by derrod** [comment, marked as answer]
No.

**Reply by KATalyzt75** [reply]
Why not? :/

**Reply by derrod** [reply]
It's in our ideas tracker: https://ideas.obsproject.com/posts/578/custom-css-for-browser-panels

It just hasn't happened yet as it's not a particularly high priority or requested feature.

**Comment by mihawk90** [comment]
Well since this is categorised as a "development question" I would guess that this question is targeting development, and the answer for that is yes, it is possible, but it would have to be written first.

The BTTV and FFZ addons are injected in the [UI/auth-twitch.cpp](https://github.com/obsproject/obs-studio/blob/5292d1ced2073810f73098aa971c9bf99cb8bc88/UI/auth-twitch.cpp#L170) so you can check how it's done there.

I'm not sure how the team was planning to implement this but I think just having a fixed location in the config directory that is loaded when present is probably the easiest way to implement it.

---

An alternative way around this limitation would be enabling FrankerFaceZ once you logged into your Twitch account in OBS.

FFZ can change the appearance of `/me` back to the old colored style under Chat > Appearance.

If you want even more customisation, in the FFZ settings under Add-Ons there is an option at the very bottom "Use Local Development Server" with which you can load your own custom FFZ Add-Ons. You could write your own that just injects your CSS from the Add-On. See FFZ's [Getting Started Guide](https://github.com/FrankerFaceZ/Add-Ons#getting-started) for how to do that.
